Add payload threshold for using uTP, and show work (#339)
ethereum/portal-network-specs: portal-wire-protocol.md

> The Portal wire protocol currently implements uTP over the `TALKREQ/TALKRESP` messages.  Future plans are to move to the [sub-protocol data transmission](https://github.com/ethereum/devp2p/issues/229) in order to use a protocol native mechanism for establishing packet streams between clients.

> Currently, the standard is to switch to uTP when the payload exceeds 1165 bytes. This may change over time, because it depends on a number of other variables. See an example derivation in rust:

Reviewer: Great for adding this in the specifications, but I believe the value should be 1177 bytes.

That's the value that gets calculated by Fluffy at least, and IIRC this was at some point in the past also looked at as reference by Trin, but perhaps it was not taken over exactly or got altered over time.

Calculation in Fluffy happens here: https://github.com/status-im/nimbus-eth1/blob/master/fluffy/network/wire/portal_protocol.nim#L123 & https://github.com/status-im/nimbus-eth1/blob/master/fluffy/network/wire/portal_protocol.nim#L404-L406

I don't have an exact test for this, but I do remember verifying this byte per byte on a encoded `talkresp` message.
I will make the `const`s more clear and can add a test for this also to show.

Reviewer: I have moved the calculation for talkreq overhead to our discv5 module and added tests there: https://github.com/status-im/nim-eth/commit/aa1e738a97d1995c1ddb821123e4842409218b0c

So I'm pretty certain that these are the correct values (at least with our discv5 impl., but that shouldn't differ really).

Author: It looks like this test is for the max size of the Talk Response payload, but the relevant payloads for a Content message are transported in a Talk Request. Does that make a difference?

Reviewer: The  content message adds just two additional bytes of overhead, so those need to be added still:

https://github.com/status-im/nimbus-eth1/blob/master/fluffy/network/wire/portal_protocol.nim#L391

You can also see that in our test vectors: https://github.com/ethereum/portal-network-specs/blob/master/portal-wire-test-vectors.md#content-response---content-payload

Author: Ok, I'll update the spec to note that we're talking about the size of the encoded message that includes the message ID, and leave out the extra 2 bytes. I don't see a test in trin for this content size max, so I'll take your values here :)

Author: Wait, but it looks like the message overhead is different for the different scenarios, like find nodes vs find content, so maybe I'll just ignore the 2 bytes here, and look at the total encoded payload size

Reviewer: Yes, it will be different for each of the wire protocol messages. But I guess the most important one here is the one that differentiates between sending content over the protocol response or over uTP

Currently, the standard is to switch to uTP when the payload exceeds 1177 bytes. This is the full encoded payload size that includes the message ID, etc. The max payload size may change over time, because it depends on a number of other variables. See an example derivation in python:
```py
# The maximum size of a Discv5 packet.
MAX_DISCV5_PACKET_SIZE = 1280

# Discv5 ordinary packet overhead
DISCV5_PACKET_OVERHEAD = (  # sum: 87
    # masking IV length
    16 +
    # static header (protocol ID || version || flag || nonce || authdata-size) length
    23 +
    # authdata length
    32 +
    # HMAC length
    16
)

# Discv5 talk request overhead
DISCV5_TALK_REQ_OVERHEAD = (  # sum: 16
    # talkResp msg id
    1 +
    # rlp encoding outer list, max length will be encoded in 2 bytes
    3 +
    # request id (max = 8) + 1 byte from rlp encoding byte string
    9 +
    # rlp encoding response byte string, max length in 2 bytes
    3
)

> # The maximum size of a portal CONTENT payload. At the time of writing, this payload either
> # corresponds to a `connection_id`, `enrs`, or `content` payload.

Reviewer: ```suggestion
# The maximum size of a portal wire protocol message payload. 
```
Considering the 2 bytes for the portal wire msg id and the Union selector (of Content response) are ignored here, we probably should phrase it like above suggestion? Which would apply then to all message requests & responses.
